examples/subscribe/subscribe.go

```diff
@@ -27,67 +28,47 @@ func main() {
 	}
 	defer c.Close()
 
-	subParams := opcua.NewDefaultSubscriptionParameters()
-	sub, err := c.Subscribe(*subParams)
+	sub, err := c.Subscribe(&opcua.SubscriptionParameters{
+		Interval: 500 * time.Millisecond,
+		Notifs:   make(chan opcua.PublishNotificationData, 37),
+	})
 	if err != nil {
 		log.Fatal(err)
 	}
+	defer sub.Cancel()
 	log.Printf("created Subscription with id %v", sub.SubscriptionID)
 
-	nodeId := ua.NewStringNodeID(uint16(*namespace), *stringId)
-
-	miCreateRequest := opcua.NewMonitoredItemCreateRequestWithDefaults(nodeId, ua.AttributeIDValue, 42)
-	res, err := c.CreateMonitoredItems(sub.SubscriptionID, ua.TimestampsToReturnBoth, miCreateRequest)
-	if err != nil {
+	miCreateRequest := opcua.NewMonitoredItemCreateRequestWithDefaults(ua.NewStringNodeID(uint16(*namespace), *stringId), ua.AttributeIDValue, 42)
+	res, err := sub.Monitor(ua.TimestampsToReturnBoth, miCreateRequest)
+	if err != nil || res.Results[0].StatusCode != ua.StatusOK {
 		log.Fatal(err)
 	}
-	for _, result := range res.Results {
-		if result.StatusCode != ua.StatusOK {
-			log.Fatal(result.StatusCode)
-		}
-	}
-	miId := res.Results[0].MonitoredItemID
-	log.Printf("created MonitoredItem with id %v", miId)
 
-	quit := time.After(30 * time.Second)
-LOOP:
+	ctx, _ := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
don't omit `cancel`. Just call `defer cancel()` out of habit.

Reply:
Wilco